### PR TITLE
Code cleanup in snap/core pkg.

### PR DESCRIPTION
**Pull request by geauxvirtual**
Summary of changes:
- Rename marshalTask and MarshalBody functions to match what the functions actually do.
- Change core/task_test.go from small to medium test as tests are reading files from filesystem.
- Rename core/task_test.go to core/task_medium_test.go.
- Create core/schedule_small_test.go and move makeSchedule tests from core/task_medium_test.go to core/schedule_small_test.go.

Testing done:
- Verified tests pass for small, medium, and legacy.

@intelsdi-x/snap-maintainers

### CODE REVIEW

File: core/task_medium_test.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+const (
+	DUMMY_FILE = "dummy.txt"
+	YAML_FILE  = "../examples/tasks/mock-file.yaml"
+	JSON_FILE  = "../examples/tasks/mock-file.json"
+	DUMMY_TYPE = "dummy"
+)
+
+func koRoutine(sch schedule.Schedule,
```
Comment (candysmurf) [inline core/task_medium_test.go]: should this named as errRoutine to be more explicit?

Comment (geauxvirtual) [inline core/task_medium_test.go]: I didn't write the function or tests in this file, just renamed the file.

Comment (candysmurf) [inline core/task_medium_test.go]: if you agree, would you please change it and we can get this merged. thanks

**Comment by candysmurf** [comment]
LGTM

**Pull request merged**